File: stack/stack.go
// some gfg exercies using stacks

package main

type Stack struct {
	top int 
	capacity uint
	array []interface
}

// Returns an initialized stack
func (stack *Stack) Init(capacity uint) *Stack {
	stack.top = -1
	stack.capacity = capacity
	stack.array = make([]interface{}, capacity)
	return stack
}

// Returns a new stack
func NewStack(capacity uint) *Stack {
	return new(Stack).Init(capacity)
}

File: workbench/stack/stack.py
class Stack:
	def __init__(self, Capacity = -1):
		self.top = -1
		self.Capacity = Capacity
		self.A = [None]*Capacity

	def push(self, data):
		if self.Capacity == self.top + 1:
			print('Stack Overfloe')
			return
		self.top += 1
		self.A[self.top] = data

	def pop(self):
	def peek(self):
	def isFull(self):
		return self.Capacity != -1
	def isEmpty(self):
		return self.Capacity == -1
